flink-core/src/main/java/org/apache/flink/api/common/operators/base/JoinOperatorBase.java

```diff
@@ -47,99 +45,29 @@
 /**
  * @see org.apache.flink.api.common.functions.FlatJoinFunction
  */
-public class JoinOperatorBase<IN1, IN2, OUT, FT extends FlatJoinFunction<IN1, IN2, OUT>> extends DualInputOperator<IN1, IN2, OUT, FT> {
-	
-	/**
-	 * An enumeration of hints, optionally usable to tell the system how exactly execute the join.
-	 */
-	public static enum JoinHint {
```

Comment:
Moving the `JoinHint` enum breaks the public API. Join hints are quite often used and I would prefer to keep them at the same location.
What do you think about renaming *your* `AbstractJoinOperatorBase` class to `JoinOperatorBase` and *your* `JoinOperatorBase` class to `InnerJoinOperatorBase`?
Alternatively, we can also move the `JoinHint` enum from `AbstractJoinOperatorBase` to `JoinOperatorBase`.